### DISCUSSION TITLE
How to disable a specfic column for sorting?

### DISCUSSION THREAD

**Discussion by KoichaDev** [TanStack/table#3536] [Q&A]
For example, I have three columns, and the specific `address` column, that is the column I want to disable the sorting functionality, but the rest can keep. How do I do that?

```js
export const MEMBERSHIP_COLUMNS = [
	{
		Header: 'Id',
		accessor: 'id',
	},
	{
		Header: 'NAME',
		accessor: 'name',
	},
	{
		Header: 'ADDRESSE',
		accessor: 'address',
	},
];

```

**Comment by kmurari103** [comment]
Try adding **disableSortBy: true** in the column headers to disable it.
ex- {
		Header: 'Id',
		accessor: 'id',
                disableSortBy: true
	},

**Reply by KoichaDev** [reply]
@kmkashyap fantastic! Thank you! That works!

What if I already added some sorting icon for that specific column I want to disable it? Example here is my code snippet that is looping through every column to display the icon. I need for example that column "id" of the header not to display the Descending, Ascending, and neutral icon, but the rest of the column can be displayed.

Here is my code snippet below

```js
... 
<thead>
	{headerGroups.map((headerGroup) => {
		return (
			<tr {...headerGroup.getHeaderGroupProps()}>
				{headerGroup.headers.map((column) => {
					return (
					<th {...column.getHeaderProps(column.getSortByToggleProps())}>
						{column.render('Header')}
						<span t={`${styles.inactive} `}>
							// Here is the icon that is sorting, but I want it not to be displayed for example on the header column "id" as an example above
							{column.isSorted ? column.isSortedDesc ? <DescendingIcon /> : <AscendingIcon /> : <NeutraIcon />}
						</span>
					</th>
				);
				})}
			</tr>
		);
	})}
</thead>

```
